ec/ec.go

```diff
@@ -0,0 +1,26 @@
+package ec
+
+import (
+	"context"
+	"time"
+
+	"github.com/filecoin-project/go-f3/gpbft"
+)
+
+type ECBackend interface {
+	// GetTipsetByEpoch should return a tipset before the one requested if the requested
+	// tipset does not exist due to null epochs
+	GetTipsetByEpoch(ctx context.Context, epoch int64) (TipSet, error)
+	GetTipset(context.Context, gpbft.TipSetKey) (TipSet, error)
+	GetHead(context.Context) (TipSet, error)
+	GetParent(context.Context, TipSet) (TipSet, error)
+
+	GetPowerTable(context.Context, gpbft.TipSetKey) (gpbft.PowerEntries, error)
+}
+
+type TipSet interface {
```

Comment:
There is a [`TipSet` type](https://github.com/filecoin-project/go-f3/blob/650861a1a3c36f9103b9010a42ce8e6b0005cb69/gpbft/chain.go#L54) already in `gpbft` package. It would be nice to unify them if they intend to covey the same entity, or rename this one to express that they are indeed different. 

This can be done in a separate PR if it would touch too many lines.

Reply:
chiming @Kubuxu in here for visibility and to give additional context. This type comes from his EC integration, and I just moved it to its own independent package.

Reply:
These are, unfortunately, _different_ tipsets.

1. This is an abstraction over the client's concept of a tipset and doesn't, e.g., carry a commitment to a power table etc.
2. The one in the `gpbft` package contains additional commitments, etc.

Honestly, we may just want to rename the one in the `gpbft` package to... ECBlock/ECBlockKey? Or something like that? But that might be even more confusing.